cpp/test/neighbors/ann_ivf_flat.cuh

```diff
@@ -154,45 +156,42 @@ class AnnIVFFlatTest : public ::testing::TestWithParam<AnnIvfFlatInputs<IdxT>> {
         cuvs::neighbors::ivf_flat::index<DataT, IdxT> idx(handle_, index_params, ps.dim);
         cuvs::neighbors::ivf_flat::index<DataT, IdxT> index_2(handle_, index_params, ps.dim);
 
-        // if (!ps.host_dataset) {
-
-        auto database_view = raft::make_device_matrix_view<const DataT, IdxT>(
-          (const DataT*)database.data(), ps.num_db_vecs, ps.dim);
-        idx = cuvs::neighbors::ivf_flat::build(handle_, index_params, database_view);
-        rmm::device_uvector<IdxT> vector_indices(ps.num_db_vecs, stream_);
-        thrust::sequence(raft::resource::get_thrust_policy(handle_),
-                         thrust::device_pointer_cast(vector_indices.data()),
-                         thrust::device_pointer_cast(vector_indices.data() + ps.num_db_vecs));
-        raft::resource::sync_stream(handle_);
-
-        IdxT half_of_data = ps.num_db_vecs / 2;
+        if (!ps.host_dataset) {
```

Comment:
raft::spatial needs to be updated. This is also bringing in `raft/spatial/knn/ann.cuh`, which shouldn't be brought over. Please update `raft::spatial` to point to `cuvs::neighbors::ivf_flat` APIs.